MFVNeutralino/python/NtupleCommon.py

```diff
@@ -302,6 +305,27 @@ def ntuple_process(settings):
     else:
         return aod_ntuple_process(settings)
 
+# Used for samples stored in inclusive miniaods; currently set up for ZH and Wplus
+# may need to change to handle different naming conventions 
+def signal_uses_random_pars_modifier(sample): 
+    to_replace = []
+
+    if sample.is_signal:
+        if sample.is_rp :
+            magic_randpar = 'randpars_filter = False'
+            decay = sample.name.split('_')[1]
+
+            ctau = float(sample.tau)/1000
+            if ctau < 1 :
+                ctau = str(ctau).replace('.', 'p')
+            else :
+                ctau = str(ctau).replace('.', 'p')
+                ctau = ctau.replace('p0', '')
```

Comment:
A comment here would be useful too--I think you set `ctau = float(sample.tau)/1000` to get ctau in mm? And then there is some formatting of the string?
(I guess the `ctau = str(ctau).replace('.', 'p') ` part is common to both the `if` and `else`, so you could move that up)

Reply:
Resolved; 
The if else comparison is needed while ctau is a float value, hence the repeated 
`ctau = str(ctau).replace('.', 'p')`